pdc/apps/unreleasedvariant/tests.py

```diff
@@ -0,0 +1,331 @@
+
+        # Try to get the module with component "bash".
+        response = self.client.get(url + '?component_name=bash', format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data.get('count'), 1)
+        self.assertEqual(response.data['results'][0]["variant_uid"], "Core")
+
+        # Try to get a module with unknown component.
+        response = self.client.get(url + '?component_name=unknown', format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data.get('count'), 0)
+
+    def test_create_and_update_unreleasedvariant(self):
+        url = reverse('unreleasedvariant-list')
+        data = {
+            'variant_id': "core", 'variant_uid': "coretest123",
+            'variant_name': "core", 'variant_version': "0",
+            'variant_release': "1", 'variant_type': 'module',
+            'variant_context': '12345678', 'koji_tag': "module-core-0-1",
+            'modulemd': 'foobar', 'active': False
+        }
+        response = self.client.post(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        url_two = reverse('unreleasedvariant-detail', args=['coretest123'])
+        data_two = {'active': True}
+        response_two = self.client.patch(url_two, data_two, format='json')
+        self.assertEqual(response_two.status_code, status.HTTP_200_OK)
+        uv = Module.objects.filter(uid='coretest123').first()
+        self.assertTrue(uv.active)
+
+    def test_create_unreleasedvariant_default_context(self):
+        url = reverse('unreleasedvariant-list')
+        data = {
+            'variant_id': "core", 'variant_uid': "coretest123",
+            'variant_name': "Core", 'variant_version': "0",
+            'variant_release': "1", 'variant_type': 'module',
+            'koji_tag': "module-core-0-1", 'modulemd': 'foobar'
+        }
+        response = self.client.post(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        uv = Module.objects.filter(uid='coretest123').first()
+        self.assertEqual(uv.context, '00000000')
+
+    def test_create_unreleasedvariant_not_default_context(self):
+        url = reverse('unreleasedvariant-list')
+        data = {
+            'variant_id': "core", 'variant_uid': "coretest123",
+            'variant_name': "Core", 'variant_version': "0",
+            'variant_release': "1", 'variant_type': 'module',
+            'variant_context': 'a23d56a8', 'koji_tag': "module-core-0-1",
+            'modulemd': 'foobar'
+        }
+        response = self.client.post(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        uv = Module.objects.filter(uid='coretest123').first()
+        self.assertEqual(uv.context, 'a23d56a8')
+
+    def test_v2_compatibility_unreleasedvariant(self):
```

Comment:
This is the only test I added to this file. The rest came from the original `pdc/apps/module/tests.py`.